keras/layers/layer.py

```diff
@@ -847,6 +847,17 @@ def maybe_convert(x):
                     "therefore destroy the mask information. Downstream "
                     "layers will not see the mask."
                 )
+        except TypeError as e:
+            if "TrackedDict" in str(e):
+                raise TypeError(
```

Comment:
Is there a way to move this to the deserialization logic? It's not robust to have it here and to rely on string matching to detect this case.

Reply:
Agreed that string matching is not reliable enough to detect this case. However, moving this to deserialization logic is difficult because we would need to parse a large nested dictionary to determine which components are not properly deserialized. In tf_keras we actually do the check this same way:

[Check for deserialization](https://github.com/keras-team/tf-keras/blob/7a85ad8c8131edb5cdddb88543d3cf8a737422ff/tf_keras/engine/base_layer.py#L2581)

Reply:
At what level would the check happen and what processing would be needed in order to be able to raise the error?

Reply:
The check would have to happen [here in the serialization_lib](https://github.com/keras-team/keras/blob/56762f99f6c1b1d7817a932c049f2885d68f4508/keras/saving/serialization_lib.py#L706C10-L706C10). However, processing-wise, we can't tell what dict within the nested dict (inner_config) actually has to be a callable. As a result, we can't error out until a symbolic call is done on that dict.

Reply:
Actually I have figured out a way to move this error handling to the serialization library via a try-except. PTAL!